src/sparc/client/services/metadata.py

```diff
@@ -0,0 +1,278 @@
+        This supports Elasticsearch JSON queries
+    list_datasets(...) : dict
+        Returns a dictionary with datasets metadata.
+    search_datasets(...) : dict
+        Returns a dictionary with datasets matching search criteria.
+
+    """
+
+    default_headers = {
+        "Content-Type": "application/json",
+        "Accept": "application/json; charset=utf-8",
+    }
+
+    host_api = "https://scicrunch.org/api/1/elastic"
+
+    scicrunch_api_key: str = None
+    profile_name: str = None
+
+    def __init__(
+        self, config: Optional[Union[dict, SectionProxy]] = None, connect: bool = False
+    ) -> None:
+        logging.info("Initializing SPARC K-Core Elasticsearch services...")
+        logging.debug(str(config))
+
+        if config is not None:
+            self.scicrunch_api_key = config.get("scicrunch_api_key")
+            logging.info("SciCrunch API Key: Found")
+            self.profile_name = config.get("pennsieve_profile_name")
+            logging.info("Profile: " + self.profile_name)
+        else:
+            logging.warning("SciCrunch API Key: Not Found")
+            logging.info("Profile: none")
+        if connect:
+            self.connect()
+
+    def connect(self) -> str:
+        """Not needed as metadata services are REST service calls"""
+        logging.info("Metadata REST services available...")
+
+        self.host_api = "https://scicrunch.org/api/1/elastic"
+        return self.host_api
+
+    def info(self) -> str:
+        """Returns information about the metadata search services."""
+
+        self.host_api = "https://scicrunch.org/api/1/elastic"
+        return self.host_api
+
+    def get_profile(self) -> str:
+        """Returns currently used API key.
+
+        Returns:
+        --------
+        A string with API Key.
+        """
+        return self.scicrunch_api_key
+
+    def set_profile(self, api_key: str) -> str:
+        """Changes the API key to the specified name.
+
+        Parameters:
+        -----------
+        api_key : str
+            The API key to use.
+
+        Returns:
+        --------
+        A string with confirmation of API key switch.
+        """
+        self.scicrunch_api_key = api_key
+        return self.scicrunch_api_key
+
+    def close(self) -> None:
+        """Not needed as metadata services are REST service calls"""
+        return self.host_api
+
+    #####################################################################
+    # Supporting Functions
+
+    #####################################################################
+    # Function to GET content from URL with retries
+    def getURL(self, url, headers="NONE"):
+        result = "[ERROR]"
+        url_session = requests.Session()
+
+        retries = Retry(
+            total=6, backoff_factor=1, status_forcelist=[403, 404, 413, 429, 500, 502, 503, 504]
+        )
+
+        url_session.mount("https://", HTTPAdapter(max_retries=retries))
+
+        success = 1
```

Comment:
maybe storing `success` as a class variable would make more sense, so that we won't repeat it in multiple places in the code?
`self.success = True?`

Reply:
Do we actually need `success` flag at all? 
Maybe we can get away with using `results={}` ?

Reply:
Can add to ticket for future updates - shouldn't hold up this release

Reply:
I disagree, it should be done now.